src/c/sk_canvas.cpp

```diff
@@ -113,6 +113,10 @@ int sk_canvas_save_layer(sk_canvas_t* ccanvas, const sk_rect_t* crect, const sk_
     return AsCanvas(ccanvas)->saveLayer(AsRect(crect), AsPaint(cpaint));
 }
 
+int sk_canvas_save_layer_rec(sk_canvas_t* ccanvas, const sk_rect_t* cbounds, const sk_paint_t* cpaint, const sk_imagefilter_t* cfilter, uint32_t flags) {
```

Comment:
Can the enum also live in the C world and then the generator can find it and do the work. This also helps because you also need to update the [src/c/sk_enums.cpp](https://github.com/mono/skia/blob/skiasharp/src/c/sk_enums.cpp) to do static checking.

Many times, the C++ api inserts a new enum or changes the order and thus we have to catch this.

Reply:
The C++ API doesn't actually use an enum but a [uint32_t typedef](https://api.skia.org/classSkCanvas.html#aa178674c37b7afc960822390abf1e91c), that's why I did it this way

Reply:
I see what you mean, but I still think we can make the C API more sensible and also get the benefit of static value checking as I know the enum values do change between releases. For example, they have re-ordered/inserted values in the SkColorType enum several times already breaking the whole system because .NET does not work the same way as headers.

Reply:
Also, since the enum is part of the SkCanvas type, to avoid conflicts when they inevatably add a different save layer rec, the name should be `sk_canvas_save_layer_rec_flags_t`.

I see in later versions of skia there are some other options available in the rec, such as filters and colorspace: https://github.com/google/skia/blob/main/include/core/SkCanvas.h#L684

So I think to avoid future confusion, the filter arg name should maybe be `backdrop`.